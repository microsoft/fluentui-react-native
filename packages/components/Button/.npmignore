# Don't include test files in package
__tests__