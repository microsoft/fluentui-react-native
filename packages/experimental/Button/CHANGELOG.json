{
  "name": "@fluentui-react-native/experimental-button",
  "entries": [
    {
      "date": "Fri, 18 Jun 2021 00:38:19 GMT",
      "tag": "@fluentui-react-native/experimental-button_v0.6.1",
      "version": "0.6.1",
      "comments": {
        "patch": [
          {
            "comment": "Apply prettier to experimental, run",
            "author": "ruaraki@microsoft.com",
            "commit": "907fea51098ba59a8235128e17fb82a327b7adb2",
            "package": "@fluentui-react-native/experimental-button"
          }
        ]
      }
    },
    {
      "date": "Tue, 15 Jun 2021 00:55:36 GMT",
      "tag": "@fluentui-react-native/experimental-button_v0.6.0",
      "version": "0.6.0",
      "comments": {
        "minor": [
          {
            "comment": "Handle ColorValue instead of string as color type",
            "author": "ruaraki@microsoft.com",
            "commit": "90e0dcb92a4ea4ee5ea5373f2c54defa127749bf",
            "package": "@fluentui-react-native/experimental-button"
          }
        ]
      }
    },
    {
      "date": "Wed, 02 Jun 2021 17:30:36 GMT",
      "tag": "@fluentui-react-native/experimental-button_v0.5.4",
      "version": "0.5.4",
      "comments": {
        "patch": [
          {
            "comment": "add useOnPressWithFocus hook",
            "author": "krsiler@microsoft.com",
            "commit": "1d9f21958a7c94c0ca3d57856f060662ae34796e",
            "package": "@fluentui-react-native/experimental-button"
          }
        ]
      }
    },
    {
      "date": "Thu, 15 Apr 2021 20:08:17 GMT",
      "tag": "@fluentui-react-native/experimental-button_v0.5.1",
      "version": "0.5.1",
      "comments": {
        "none": [
          {
            "comment": "Merge remote-tracking branch 'upstream/master' into amgleitman/icon-macos",
            "author": "adam.gleitman@gmail.com",
            "commit": "06834001f0f0238d2a7ad99a51a430b28eaa8daa",
            "package": "@fluentui-react-native/experimental-button"
          }
        ]
      }
    },
    {
      "date": "Sat, 06 Mar 2021 00:20:05 GMT",
      "tag": "@fluentui-react-native/experimental-button_v0.5.0",
      "version": "0.5.0",
      "comments": {
        "patch": [
          {
            "comment": "Base theme on macOS changed to apple theme",
            "author": "saadnajmi2@gmail.com",
            "commit": "1b3d9f6ef10c1138b6f9701b5edf9ef52be623e6",
            "package": "@fluentui-react-native/experimental-button"
          },
          {
            "comment": "Use ColorValue from @fluentui-react-native/theme-types",
            "author": "saadnajmi2@gmail.com",
            "commit": "71b29ea2c7c28d3cdadbade15f2b2c60127e48ec",
            "package": "@fluentui-react-native/experimental-button"
          }
        ],
        "minor": [
          {
            "comment": "Update to react-native 0.63",
            "author": "30809111+acoates-ms@users.noreply.github.com",
            "commit": "d35d3b45f12a670648edaa7143b210ed43360996",
            "package": "@fluentui-react-native/experimental-button"
          }
        ]
      }
    },
    {
      "date": "Mon, 01 Mar 2021 22:53:12 GMT",
      "tag": "@fluentui-react-native/experimental-button_v0.4.2",
      "version": "0.4.2",
      "comments": {
        "patch": [
          {
            "comment": "Update deps",
            "author": "warleu@microsoft.com",
            "commit": "23cd3d437e63e105f5feee16a1b8ef8e769b61c1",
            "package": "@fluentui-react-native/experimental-button"
          }
        ]
      }
    },
    {
      "date": "Wed, 13 Jan 2021 20:04:26 GMT",
      "tag": "@fluentui-react-native/experimental-button_v0.4.0",
      "version": "0.4.0",
      "comments": {
        "minor": [
          {
            "comment": "fix depcheck error by adding component-cache dependency for experimental button",
            "author": "67026167+chiuam@users.noreply.github.com",
            "commit": "a0b6666ced16ccb8d06e7f91dc3136294d209de7",
            "package": "@fluentui-react-native/experimental-button"
          }
        ]
      }
    },
    {
      "date": "Wed, 06 Jan 2021 18:41:50 GMT",
      "tag": "@fluentui-react-native/experimental-button_v0.3.3",
      "version": "0.3.3",
      "comments": {
        "patch": [
          {
            "comment": "Bump react-native-win32",
            "author": "krsiler@microsoft.com",
            "commit": "07395b1141d73a37a8d9ccd09bafadd205b5a528",
            "package": "@fluentui-react-native/experimental-button"
          }
        ]
      }
    },
    {
      "date": "Thu, 17 Dec 2020 00:09:27 GMT",
      "tag": "@fluentui-react-native/experimental-button_v0.3.2",
      "version": "0.3.2",
      "comments": {
        "none": [
          {
            "comment": "Add readme for svg",
            "author": "warleu@microsoft.com",
            "commit": "c2c28c356aabbd3e74a8d085015a3f4507fbdbe9",
            "package": "@fluentui-react-native/experimental-button"
          }
        ]
      }
    },
    {
      "date": "Fri, 23 Oct 2020 22:27:37 GMT",
      "tag": "@fluentui-react-native/experimental-button_v0.3.0",
      "version": "0.3.0",
      "comments": {
        "minor": [
          {
            "comment": "RNIcon feature",
            "author": "warleu@microsoft.com",
            "commit": "83a0f448bf9ee9dddece08e7c277304dcef96388",
            "package": "@fluentui-react-native/experimental-button"
          }
        ]
      }
    },
    {
      "date": "Thu, 22 Oct 2020 21:21:59 GMT",
      "tag": "@fluentui-react-native/experimental-button_v0.2.1",
      "version": "0.2.1",
      "comments": {
        "none": [
          {
            "comment": "onDismiss not called when contextual menu is dismissed with item click",
            "author": "lehon@microsoft.com",
            "commit": "4f3ed9d5555e86a6017901aa2f010732d6d6fe5a",
            "package": "@fluentui-react-native/experimental-button"
          }
        ]
      }
    },
    {
      "date": "Mon, 19 Oct 2020 22:06:02 GMT",
      "tag": "@fluentui-react-native/experimental-button_v0.2.1",
      "version": "0.2.1",
      "comments": {
        "patch": [
          {
            "comment": "rework theme usage in tester",
            "author": "jasonmo@microsoft.com",
            "commit": "5657a340608ad2e1fb6b7e61f92d706e350847b1",
            "package": "@fluentui-react-native/experimental-button"
          }
        ]
      }
    },
    {
      "date": "Mon, 12 Oct 2020 04:09:31 GMT",
      "tag": "@fluentui-react-native/experimental-button_v0.2.0",
      "version": "0.2.0",
      "comments": {
        "minor": [
          {
            "comment": "new variants",
            "author": "email not defined",
            "commit": "d1c543f21301b22a84fb6968a2c2612bb487be39",
            "package": "@fluentui-react-native/experimental-button"
          }
        ]
      }
    }
  ]
}
