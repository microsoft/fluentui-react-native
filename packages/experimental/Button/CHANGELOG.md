# Change Log - @fluentui-react-native/experimental-button

This log was last generated on Mon, 12 Oct 2020 04:09:31 GMT and should not be manually modified.

<!-- Start content -->

## 0.2.0

Mon, 12 Oct 2020 04:09:31 GMT

### Minor changes

- new variants (email not defined)
