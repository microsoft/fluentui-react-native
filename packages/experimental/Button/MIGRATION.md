# Button Migration

## STATUS: WIP ðŸš§

This Migration guide is a work in progress and is not yet ready for use.

## Migration from v0 Button

### Component renames

Primary and Stealth buttons now map to `Button`:

| v0 `Button`                             | v1 `Button`                                          |
| --------------------------------------- | ---------------------------------------------------- |
| `<Button text="Hello, world" />`        | `<Button>Hello, world</Button>`                      |
| `<PrimaryButton text="Hello, world" />` | `<Button appearance='primary'>Hello, world</Button>` |
| `<StealthButton text="Hello, world" />` | `<Button appearance='subtle'>Hello, world</Button>`  |

### Props that remain as is

- `children`
- `icon`
- `onClick`
- `testID`
- `tooltip`

### Props no longer supported with an equivalent functionality in v1 Button

- `componentRef` => Use `ref` instead
- `content` => Pass the content as `children` instead
- `endIcon` => Use `after` value for `iconPosition` prop and pass icon information into `icon` prop instead
- `startIcon` => Use `before` value for `iconPosition` prop and pass icon information into `icon` prop instead

### Props no longer supported without an equivalent functionality in v1 Button

- Cannot use both `startIcon` and `endIcon` at the same time in v1

## Migration from FluentUI v9 Button

### Props that remain as is

- `children`
- `disabled`
- `icon`
- `onClick`
- `testID`
- `tooltip`

### Props renamed to align with ReactNative

No `Button` specific renames. See [this porting guide](../../../docs/pages/Guides/PortingFromFluentUI.md) for general guidance.

### Prop differences due to technical differences and limitations

- `icon` takes in a props object instead of the JSX element itself. This is due to framework differences from FluentUI.
- `iconOnly` must be supplied for components do not have any text content for them to be styled correctly. This is due to framework differences from FluentUI.

### Other Prop differences

- `appearance` does not include values for `transparent` or `outline` as neither currently are used by native platforms.
- `iconPosition` uses `before` and `after` as values instead of `left` and `right` for better right to left support.

## Property mapping

| v0 `Button`    | v1 `Button`    |
| -------------- | -------------- |
| `componentRef` | `ref`          |
| `content`      |                |
| `endIcon`      | `iconPosition` |
| `icon`         | `icon`         |
| `onClick`      | `onClick`      |
| `statIcon`     | `iconPosition` |
| `tooltip`      | `tooltip`      |
