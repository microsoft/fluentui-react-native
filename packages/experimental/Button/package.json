{
  "name": "@fluentui-react-native/experimental-button",
  "version": "0.1.0",
  "description": "A cross-platform Button component using the Fluent Design System",
  "main": "src/index.ts",
  "module": "src/index.ts",
  "typings": "lib/index.d.ts",
  "private": true,
  "onPublish": {
    "main": "lib-commonjs/index.js",
    "module": "lib/index.js"
  },
  "scripts": {
    "build": "fluentui-scripts build",
    "clean": "fluentui-scripts clean",
    "just": "fluentui-scripts",
    "lint": "fluentui-scripts eslint",
    "test": "fluentui-scripts jest",
    "update-snapshots": "fluentui-scripts jest -u",
    "verify-api": "fluentui-scripts verify-api-extractor",
    "update-api": "fluentui-scripts update-api-extractor"
  },
  "dependencies": {
    "@fluentui-react-native/adapters": ">=0.3.34 <1.0.0",
    "@fluentui-react-native/experimental-framework": "0.2.2",
    "@fluentui-react-native/interactive-hooks": ">=0.4.38 <1.0.0",
    "@fluentui-react-native/experimental-text": ">=0.1.0 <1.0.0",
    "@fluentui-react-native/tokens": ">=0.5.1 <1.0.0"
  },
  "devDependencies": {
    "@office-iss/react-native-win32": "0.62.0-preview.3",
    "@types/react-native": "^0.62.0",
    "@uifabricshared/build-native": "^0.1.1",
    "@uifabricshared/eslint-config-rules": "^0.1.1",
    "react-native": "0.62.2"
  },
  "peerDependencies": {
    "react-native": ">=0.60.0"
  },
  "author": "",
  "license": "MIT"
}
