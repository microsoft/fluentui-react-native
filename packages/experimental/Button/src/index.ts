export * from './Button.types';
export * from './Button';
export * from './CompoundButton';
export * from './ToggleButton';
