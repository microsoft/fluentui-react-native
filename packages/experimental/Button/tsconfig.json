{
  "extends": "@fluentui-react-native/scripts/tsconfig.json",
  "compilerOptions": {
    "importHelpers": true,
    "outDir": "lib",
    "types": ["node"]
  },
  "include": ["src"]
}
