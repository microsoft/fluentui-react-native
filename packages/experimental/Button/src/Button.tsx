/** @jsx withSlots */
import * as React from 'react';
import { Image, View } from 'react-native';
import { ButtonProps, buttonName, ButtonType } from './Button.types';
import { Text } from '@fluentui-react-native/experimental-text';
import { stylingSettings } from './Button.styling';
import { filterImageProps } from '@fluentui-react-native/adapters';
import { compose, mergeProps, withSlots, UseSlots } from '@fluentui-react-native/framework';
import { useButton } from './useButton';

export const Button = compose<ButtonType>({
  displayName: buttonName,
  ...stylingSettings,
  slots: {
    root: View,
    icon: Image,
    content: Text,
  },
  filters: {
    icon: filterImageProps,
  },
  render: (userProps: ButtonProps, useSlots: UseSlots<ButtonType>) => {
    const button = useButton(userProps);
    // grab the styled slots
    const Slots = useSlots(userProps, (layer) => button.state[layer] || userProps[layer]);
    // now return the handler for finishing render
    return (final: ButtonProps, ...children: React.ReactNode[]) => {
      const { icon, content, ...mergedProps } = mergeProps(button.props, final);

      return (
        <Slots.root {...mergedProps}>
          {icon && <Slots.icon key="icon" source={{ uri: icon }} />}
          {content && <Slots.content key="content">{content}</Slots.content>}
          {children}
        </Slots.root>
      );
    };
  },
});

export default Button;
