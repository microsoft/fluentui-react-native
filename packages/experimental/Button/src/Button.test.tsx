import * as React from 'react';
import { Button } from './Button';
import * as renderer from 'react-test-renderer';
import { checkRenderConsistency, checkReRender } from '@fluentui-react-native/test-tools';

describe('Button component tests', () => {
  it('Button default', () => {
    const tree = renderer.create(<Button content="Default Button" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Button simple rendering does not invalidate styling', () => {
    checkRenderConsistency(() => <Button content="Default button" />, 2);
  });

  it('Button re-renders correctly', () => {
    checkReRender(() => <Button content="Render twice" />, 2);
  });

  it('Button shares produced styles across multiple renders', () => {
    const style = { backgroundColor: 'black' };
    checkRenderConsistency(() => <Button content="Shared styles" style={style} />, 2);
  });

  it('Button re-renders correctly with style', () => {
    const style = { borderColor: 'blue' };
    checkReRender(() => <Button content="Shared Style Render" style={style} />, 2);
  });
});
