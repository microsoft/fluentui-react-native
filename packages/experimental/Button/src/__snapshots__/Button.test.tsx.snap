// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component tests Button circular 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onPress={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  shape="circular"
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#f3f2f1",
      "borderColor": "#8a8886",
      "borderRadius": 9999,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "minWidth": 96,
      "padding": 5,
      "paddingHorizontal": 15,
      "width": undefined,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#323130",
        "fontFamily": "Segoe UI",
        "fontSize": 14,
        "fontWeight": "600",
        "margin": 0,
        "marginBottom": 0,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
      }
    }
  >
    Circular Button
  </Text>
</View>
`;

exports[`Button component tests Button default 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onPress={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#f3f2f1",
      "borderColor": "#8a8886",
      "borderRadius": 4,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "minWidth": 96,
      "padding": 5,
      "paddingHorizontal": 15,
      "width": undefined,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#323130",
        "fontFamily": "Segoe UI",
        "fontSize": 14,
        "fontWeight": "600",
        "margin": 0,
        "marginBottom": 0,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
      }
    }
  >
    Default Button
  </Text>
</View>
`;

exports[`Button component tests Button large 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onPress={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size="large"
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#f3f2f1",
      "borderColor": "#8a8886",
      "borderRadius": 4,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "minWidth": 96,
      "padding": 7,
      "paddingHorizontal": 15,
      "width": undefined,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#323130",
        "fontFamily": "Segoe UI",
        "fontSize": 16,
        "fontWeight": "600",
        "margin": 0,
        "marginBottom": 0,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
      }
    }
  >
    Large Button
  </Text>
</View>
`;

exports[`Button component tests Button primary 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  appearance="primary"
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onPress={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#0078d4",
      "borderColor": "#005a9e",
      "borderRadius": 4,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "minWidth": 96,
      "padding": 5,
      "paddingHorizontal": 15,
      "width": undefined,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontFamily": "Segoe UI",
        "fontSize": 14,
        "fontWeight": "600",
        "margin": 0,
        "marginBottom": 0,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
      }
    }
  >
    Primary Button
  </Text>
</View>
`;

exports[`Button component tests Button small 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onPress={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size="small"
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#f3f2f1",
      "borderColor": "#8a8886",
      "borderRadius": 4,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "minWidth": 64,
      "padding": 3,
      "paddingHorizontal": 11,
      "width": undefined,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#323130",
        "fontFamily": "Segoe UI",
        "fontSize": 12,
        "fontWeight": "400",
        "margin": 0,
        "marginBottom": 0,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
      }
    }
  >
    Small Button
  </Text>
</View>
`;

exports[`Button component tests Button square 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onPress={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  shape="square"
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#f3f2f1",
      "borderColor": "#8a8886",
      "borderRadius": 0,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "minWidth": 96,
      "padding": 5,
      "paddingHorizontal": 15,
      "width": undefined,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#323130",
        "fontFamily": "Segoe UI",
        "fontSize": 14,
        "fontWeight": "600",
        "margin": 0,
        "marginBottom": 0,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
      }
    }
  >
    Square Button
  </Text>
</View>
`;

exports[`Button component tests Button subtle 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  appearance="subtle"
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onPress={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#ffffff",
      "borderColor": "#ffffff",
      "borderRadius": 4,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "center",
      "minWidth": 96,
      "padding": 5,
      "paddingHorizontal": 15,
      "width": undefined,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#323130",
        "fontFamily": "Segoe UI",
        "fontSize": 14,
        "fontWeight": "600",
        "margin": 0,
        "marginBottom": 0,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
      }
    }
  >
    Subtle Button
  </Text>
</View>
`;
